src/app.ts

```diff
@@ -49,12 +54,73 @@ const getAppInfo = (() => {
   }
 })()
 
+async function parseAuthUrl(forwardedUrl) {
+  const appInfo = await getAppInfo()
+  const query = parseUrl(forwardedUrl).query
+  const authResponse = parseQueryString(query)
+
+  if (authResponse.state !== appInfo.state) {
+	console.error('Auth state mismatch')
+	throw new Error('Auth state mismatch')
+  }
+
+  const authCode = authResponse.code
+  debug('authcode =>', authCode)
+
+  try {
+	const { accessToken, refreshToken } = await getAccessToken(
+	  appInfo,
+	  authCode
+	)
+
+	debug('token =>', accessToken)
+	if (await verifyAccess(accessToken)) {
+	  await saveCode('access_token', accessToken)
+	  await saveCode('refresh_token', refreshToken)
+
+	  mainWindow.goToMonux()
+	} else {
+	  console.error('Invalid access token')
+	  throw new Error('Invalid access token')
+	}
+  } catch (err) {
+	console.error(err)
+	throw new Error(err)
+  }
+}
+
+const isSecondInstance = app.makeSingleInstance(async (commandLine, workingDirectory) => {
+  // Someone tried to run a second instance, we should focus our window.
+  if (mainWindow) {
+    if (mainWindow.hasWindow) mainWindow.focus()
+    mainWindow.focus()
+    if( process.platform === 'win32' && commandLine.length > 1) {
+	  var urlParams = commandLine.filter(function(param){
```

Comment:
`.filter` could be replaced with a `.find` and avoid the array accesses later.

Eg.

```
const authUrl = commandLine.filter((param) => {
  return param.startsWith('monux://')
})

if (authUrl) {
  await parseAuthUrl(urlParams[0])
} else {
  console.err('Auth url not found')
  throw new Error('Auth url not found')
}
```